source/UF/Replacement.lagda

```diff
@@ -0,0 +1,107 @@
+Ian Ray, 17th August 2025.
+
+The type theoretic axiom of replacement (section 24.5 of Introduction to
+Homotopy Type Theory by Egbert Rijke:
+https://ulrikbuchholtz.dk/hott1920/hott-intro.pdf)
+is a statement about the size of the image of a function when certain smallness
+assumptions are imposed on the domain and codomain. The nomenclature is derived
+from the set theoretic axiom of replacement, where 'sethood' provides a notion
+of size. In type theory, this statement may be assumed or proven depending on
+the context. Thus, we will call it 'type replacement' or simply 'replacement'
+(when there is no risk of confusion with set replacement).
+
+The statement of type replacement is as follows:
+The image of a map f : A → X, from a small type A to a locally small type X, is
+itself small.
+
+Note: some authors use the term 'essentially small' for what the TypeTopology
+refers to as simply 'small'. Although, often times it is desirable to explicitly
+state the universe with which a type is small relative to, as you will see in
+the code below.
+
+Type replacement is provable in the presence of a certain class of higher
+inductive types (HITs). In particular, "The Join Construction" by Egbert Rijke
+(https://arxiv.org/abs/1701.07538.) provides a construction that allows a proof
+of type replacement in the presence of 'graph quotients'. More conservativly one
+may carry out this construction merely with pushouts (in fact, one only requires
+the join of maps and sequential colimits, which are instances of pushouts).
+This route is actively being explored in other TypeTopology files.
+
+It is worth noting that the status of type replacement in the hierarchy of HIT
```

Comment:
What does hierarchy of HIT strength mean when type replacement is not itself a HIT?